BSPConvert.Lib/Source/Shader.cs

```diff
@@ -146,13 +146,14 @@ public class ShaderStage
 		public TextureBundle[] bundles = new TextureBundle[NUM_TEXTURE_BUNDLES]; // Path to image file
 		public ShaderStageFlags flags;
 
-		public WaveForm rgbWave;
+		public WaveForm rgbWave = new WaveForm();
 		public ColorGen rgbGen;
 
-		public WaveForm alphaWave;
+		public WaveForm alphaWave = new WaveForm();
 		public AlphaGen alphaGen;
 
 		public byte[] constantColor = new byte[4];
+		public float constantAlpha = 1.0f;
```

Comment:
Remove this. Use `constantColor[3]` for storing alpha instead.

Reply:
Also you have to multiply the value by 255 when reading from the shader file since `constantColor` uses bytes instead of floats.

Reply:
See https://github.com/id-Software/Quake-III-Arena/blob/dbe4ddb10315479fc00086f08e25d968b4b43c49/code/renderer/tr_shader.c#L889